### DISCUSSION TITLE
Is it possible to define windows' sizes when they change from tiled to floating ?

### DISCUSSION THREAD

**Discussion by Aqa-Ib** [hyprwm/Hyprland#5246] [Q&A]
```
windowrulev2 = size 90% 80%, floating:1
windowrulev2 = center 1, floating:1
```
That rules only apply when windows are initially opened floating. 

Is it somehow possible to apply these rules to windows when their state change from tiled to floating ?

Thanks for the attention.

EDIT: OK, I see that the IPC have 'changefloatingmode'. I am trying to script this one. I will let you know if I get it working.

**Comment by Aqa-Ib** [comment]
Alright, I got it but it is buggy. Sometimes it will not resize the window, I do not know why. Whatever. If anyone knows why please let me know. Thanks.

```
#!/bin/sh

handle() {
  case $1 in
    changefloatingmode*,1) hyprctl dispatch resizeactive exact 90% 80%; hyprctl dispatch centerwindow 1 ;;
  esac
}

socat -U - UNIX-CONNECT:/tmp/hypr/$HYPRLAND_INSTANCE_SIGNATURE/.socket2.sock | while read -r line; do handle "$line"; done
```

**Reply by Aqa-Ib** [reply]
It seems like it needs a "sleep" before the hyprctl commands for whatever reason to work without being buggy. Thoughts ?

**Reply by Aqa-Ib** [reply]
```
changefloatingmode*,1) sleep 0.000000000000000000000000000001; hyprctl dispatch resizeactive exact 90% 80%; hyprctl dispatch centerwindow 1 ;;
```
This is working without bugs. I don't know how many zeroes are needed there, but whatever, it just works.
If anyone knows a cleaner way to do this, please let me know.

**Comment by udaemon** [comment]
Hey, I seem to have found another simple solution. In hyprland.conf I've put:

bind = $mainMod SHIFT, V, exec, hyprctl --batch "dispatch togglefloating ; dispatch resizeactive exact 80% 80% ; dispatch centerwindow 1"

and together with the default bind for float windows I have a handy pair. :)
may it help!
